patches/tModLoader/Terraria/DataStructures/ItemCreationContext.TML.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Terraria.DataStructures;
+
+// Various additional ItemCreationContext classes for tML purposes.
+
+/// <summary>
+/// Created in the context of a player buying an item from a shop. Very similar to <seealso cref="RecipeItemCreationContext"/> in
+/// functionality.
+/// </summary>
+public class BuyItemCreationContext : ItemCreationContext
+{
+	/// <summary>
+	/// An item stack that the bought item will be combined with (via OnStack).
+	/// </summary>
+	public Item DestinationStack;
+
+	/// <summary>
+	/// The NPC that this item was bought from.
+	/// </summary>
+	public NPC NPCBoughtFrom;
+
+	/// <summary>
+	/// The player that bought this item.
+	/// </summary>
+	public Player PlayerCustomer;
```

Comment:
No need for this. Only runs on local clients anyway, so `Main.LocalPlayer` is sufficient. Aim for parity with `JourneyDuplicationItemCreationContext` and `RecipeItemCreationContext`

Perhaps `JourneyDuplicationItemCreationContext` should get a `DestinationStack` parameter?